changes/2020-07-17_cache-entry-identifier-formulas/change.md

```diff
@@ -0,0 +1,177 @@
+they use to compute these identifiers,
+and this change adds those formulas to the specification.
+
+## Out of Scope
+
+- Changing the shape of the caching CMM operations is out of scope.
+- Changing the shape of the CMC interface operations is out of scope.
+
+## Motivation
+
+In all generally-available implementations of the AWS Encryption SDK (ESDK),
+the caching Cryptographic Materials Manager (caching CMM)
+uses a particular set of formulas
+to compute the identifiers for cache entries when interacting with the Cryptographic Materials Cache (CMC).
+Although these implementations agree on this set of formulas,
+the formulas are neither explicitly stated nor mandated in the specification.
+
+Mandating that implementations share cache entry identifier formulas
+has a number of benefits:
+
+1.  Implementers will have a definitive reference
+    for precisely which cache entry fields MUST be used to distinguish cache entries,
+    and also provides a concrete data format by which to serialize them.
+1.  It is easier to reason about the caching CMM across implementations
+    if one can assume that they distinguish cache entries by the same criteria.
+1.  If it becomes necessary to change cache identifiers in the future
+    (e.g., in order to distinguish cache entries by more features),
+    then we can do so uniformly across implementations.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+We MUST add ESDK test vectors to verify that caching CMM implementations
+correctly implement the cache entry identifier formulas.
+
+## Guide-level Explanation
+
+See [Reference-level Explanation](#reference-level-explanation).
+
+## Reference-level Explanation
+
+### Preliminaries
+
+Each of the cache entry identifier formulas includes a serialized encryption context,
+as defined in the
+[Key Value Pairs specification](../../data-format/message-header.md#key-value-pairs).
+In the following sections we use `SerializeEncryptionContext`
+to denote the function that,
+given an encryption context,
+returns the serialization of the encryption context.
+
+Some of the cache entry identifier formulas include
+the two-byte algorithm suite ID for the algorithm suite in a materials request.
+The algorithm suite IDs are defined in the
+[Supported Algorithm Suites specification](../../framework/algorithm-suites.md#supported-algorithm-suites).
+In the following sections we use `AlgorithmSuiteId`
+to the denote the function that,
+given an algorithm suite as specified in a materials request,
+returns the corresponding two-byte algorithm suite ID.
+
+### Encryption Materials, Without Algorithm Suite
+
+If the Get Encryption Materials request does not specify an algorithm suite,
+then the cache entry identifier MUST be calculated
+as the SHA-512 hash of the concatenation of the following byte strings,
+in the order listed:
+
+1. The SHA-512 hash of the caching CMM’s Partition ID
```

Comment:
The partition ID is a string.
To hash we need bytes.
This is specifically the SHA-512 hash
of the string encoded as utf-8 bytes.